core/src/verify.rs

```diff
@@ -246,6 +248,28 @@ impl CommitSequenceVerifier {
                 "consensus_delegatee cannot be changed".to_string(),
             ));
         }
+
+        let previous_version = Version::parse(&self.reserved_state.version);
+        let new_version = Version::parse(&rs.version);
+        
+        match (previous_version, new_version) {
+            (Ok(previous_version), Ok(new_version)) => {
+                if previous_version != new_version {
+                    let requirement =
+                        VersionReq::parse(&format!("> {}", previous_version)).unwrap();
```

Comment:
I think rather than utilizing `unwrap` it would be a much more graceful way to manage the potential panic:
```suggestion
if previous_version != new_version {
            match VersionReq::parse(&format!("> {}", previous_version)) {
                Ok(requirement) => {
                    if !requirement.matches(&new_version) {
                        return Err(Error::InvalidArgument(format!(
                            "Version advance is incorrect"
                        )));
                    }
                }
```

Reply:
I'll remember that using unwrap can lead to a panic if the Result is an Err, which is not ideal for error handling.